#373 Adds ability to automate developer tasks and ci flows

betarelease opened the pull request:
- adds a command chain to run the ci process locally before commit/PR
<!--

Thank you for participating with our effort to build a more secure software supply chain.
Before submitting your Pull Request please check the following.

-->

## PR Checklist

<!--

Locally run the build process

-->
- [x ] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
- [ x] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
- [ x] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
- [ x] I've not introduced any new known security vulnerabilities. I've run `cargo audit`

<!--

Make certain your Pull Request has the following.

-->
- [x] I've included a brief description and a good title of the proposed changes and how to test them.
- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
- [x] I've checked that the code is up-to-date with the `pyrsia/main` branch.
- [x] I've read ["What is a Good PR?"](good_pr.md)
- [x] I've assigned this Pull Request to "pyrsia/collaborators"

## Description

Fixes pyrsia/pyrsia#373 
Adds ability to automate developer tasks 
- adds a command chain to run the ci process locally before commit/PR

## Screenshots (optional)

johanvos commented on src/docker/v2/models/blobs.rs:
```diff
@@ -0,0 +1,118 @@
+/*
```
This is mainly a duplication of handlers/blobs.rs ?

betarelease replied:
You are correct. I actually built this PR on top of my other PR which was still draft. So a good idea maybe to just create this PR independently. I will do that. 

And the duplication you noticed is correct. I have not removed code from original location yet. I will do that as part of the other PR.

johanvos commented on readme.md:
```diff
@@ -92,11 +92,35 @@ For now Pyrsia only supports Docker artifacts. Follow these steps to run a Pyrsi
-    Connected Peers Count:   1
+    Connected Peers Count:   0
+    Artifacts Count:         12 # reflects the number of artifacts that the pyrsia_node has stored on the network
+    Total Disk Available:    983112
```
if we mention this, there should always be a unit. KB? MB? blocks?...

betarelease replied:
yes. I did not make this particular change. But will add that feedback.

prince-chrismc commented:
Please rebase the PR, the diff is a little confusing 🙏

prince-chrismc commented:
Please remove the issue number from the title 🙏 it's repeated several times so we can leave it in the description

prince-chrismc commented:
You should update the PR template with the new command(s) that contributor should be running

betarelease commented:
I made a mistake by creating this PR on top of a draft PR. I will rebase with main and resend this PR. Sorry for the trouble. Although if you still have feedback please share that here and I will fix those items.

This was limited for cargo-make change and Makefile.toml

betarelease commented:
Created #393 which isolates the code specific to this issue and addresses some concerns expressed here. 

Since this PR was from a bad branch and not from rebased main - closing this one.
